### ISSUE TITLE
Add cppcheck to CI?

### ISSUE THREAD

**Issue by timsnyder** [sparcians/map#117]
@klingaard do you have any interest in running cppcheck on c++ code in MAP during the CI?   I was noticing that the MAP CMakeLists.txt doesn't ever find cppcheck for me and then looked at how the internal project is doing it.  I could make it part of the CI for MAP at some point if you want it.

This is just something easy that I could do in the background.  I know it's not important.

**Comment by klingaard** [comment]
No, it’s important.  I appreciate your offer, but we’ll hold off for a little bit longer.  There are a few items that need to be addressed before we have a clean cppcheck.  It’s on my list of things to do.

**Comment by timsnyder** [comment]
I'm happy to set it up so that it runs on master to generate the report but doesn't block PRs from getting in.

**Comment by timsnyder** [comment]
I see that the conda package doesn't include `cppcheck-htmlreport`.  I'll get it added in https://github.com/conda-forge/cppcheck-feedstock/issues/20

**Comment by timsnyder** [comment]
The maintainer of the cppcheck package added `cppcheck-htmlreport`.  The only thing left to make it work is to also add cmake configure files to the cppcheck conda package or copy over the cmake find module that dabble is using into MAP.
